pkg/test/utils/kubernetes.go

```diff
@@ -667,23 +677,23 @@ func NewClusterRoleBinding(apiVersion, kind, name, namespace string, serviceAcco
 }
 
 // NewPod creates a new pod object.
-func NewPod(name, namespace string) runtime.Object {
+func NewPod(name, namespace string) *unstructured.Unstructured {
 	return NewResource("v1", "Pod", name, namespace)
 }
 
 // WithNamespace naively applies the namespace to the object. Used mainly in tests, otherwise
 // use Namespaced.
-func WithNamespace(obj runtime.Object, namespace string) runtime.Object {
-	obj = obj.DeepCopyObject()
+func WithNamespace(obj *unstructured.Unstructured, namespace string) *unstructured.Unstructured {
+	copy := obj.DeepCopy()
 
-	m, _ := meta.Accessor(obj) //nolint:errcheck // runtime.Object don't have the error issues of interface{}
+	m, _ := meta.Accessor(copy) //nolint:errcheck // runtime.Object don't have the error issues of interface{}
```

Comment:
I think the //nolint:errcheck is obsolete now, do we still need this? If yes, we should at least adjust the comment to `unstructured.Unstructured`

Reply:
still needed...